signingscript/sign.py

```diff
@@ -422,13 +428,68 @@ def _get_widevine_signing_files(file_list):
             fmt = 'widevine'
         if fmt:
             log.debug("Found {} to sign {}".format(filename, fmt))
-            if "{}.sig".format(filename) not in file_list:
+            sigpath = _get_mac_sigpath(filename)
+            if sigpath not in file_list:
                 files[filename] = fmt
             else:
                 log.debug("{} is already signed! Skipping...".format(filename))
     return files
 
 
+# _run_generate_precomplete {{{1
+def _run_generate_precomplete(context, tmp_dir):
+    """Regenerate `precomplete` file with widevine sig paths for complete mar."""
+    log.info("Generating `precomplete` file...")
+    path = _ensure_one_precomplete(tmp_dir, "before")
+    with open(path, "r") as fh:
+        before = fh.readlines()
+    generate_precomplete(os.path.dirname(path))
+    path = _ensure_one_precomplete(tmp_dir, "after")
+    with open(path, "r") as fh:
+        after = fh.readlines()
+    # Create diff file
+    diff_path = os.path.join(context.config['work_dir'], "precomplete.diff")
+    with open(diff_path, "w") as fh:
+        for line in difflib.ndiff(before, after):
+            fh.write(line)
+    utils.copy_to_dir(
+        diff_path, context.config['artifact_dir'],
+        target="public/logs/precomplete.diff"
```

Comment:
Minor issue: The file doesn't display in the browser. We should fix this in https://github.com/mozilla-releng/scriptworker/blob/65b110f21fed199268062191ec9cb34a2bf878ea/scriptworker/artifacts.py#L27